src/main/java/net/pms/newgui/components/CustomJSpinner.java

```diff
@@ -0,0 +1,315 @@
+				((SpinnerIntModel) model).getStepSize()
+			));
+		}
+
+		if (enterMoveFocus) {
+			if (this.getEditor() instanceof CustomJSpinner.IntegerEditor) {
+				((CustomJSpinner.IntegerEditor) this.getEditor()).getTextField().addKeyListener(new KeyListener() {
+					@Override
+					public void keyTyped(KeyEvent e) {}
+					@Override
+					public void keyReleased(KeyEvent e) {}
+					@Override
+					public void keyPressed(KeyEvent e) {
+						if (e.getKeyCode() == KeyEvent.VK_ENTER) {
+							FocusManager.getCurrentManager().focusNextComponent();
+						}
+					}
+				});
+			}
+		}
+	}
+
+	protected JComponent createEditor(SpinnerModel model) {
+		if (model instanceof SpinnerDateModel) {
+			return new DateEditor(this);
+		} else if (model instanceof SpinnerNumberModel) {
+			return new NumberEditor(this);
+		} else if (model instanceof SpinnerListModel) {
+			return new ListEditor(this);
+		} else if (model instanceof SpinnerIntModel) {
+			return new IntegerEditor(this);
+		} else {
+			return new DefaultEditor(this);
+		}
+	}
+
+	public JToolTip createToolTip() {
+	    JToolTip tip = new HyperLinkToolTip();
+	    tip.setComponent(this);
+	    return tip;
+	}
+
+	protected class MouseWheelRoll  implements MouseWheelListener {
+
+		private int minimum, maximum, stepSize;
+		private CustomJSpinner spinner;
+
+		private MouseWheelRoll(CustomJSpinner spinner, int minimum, int maximum, int stepSize) {
+			this.spinner = spinner;
+			this.minimum = minimum;
+			this.maximum = maximum;
+			this.stepSize = stepSize;
+		}
+
+		@Override
+		public void mouseWheelMoved(MouseWheelEvent e) {
+	        if (e.getScrollType() != MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+	            return;
+	        }
+
+	        int value = (int) spinner.getValue();
+	        value -= e.getWheelRotation() * stepSize;
+	        if (value == minimum + stepSize && minimum % stepSize != 0) {
+	        	value = ((minimum / stepSize) * stepSize) + stepSize;
+	        } else if (value == maximum - stepSize && maximum % stepSize != 0) {
+	        	value = (maximum / stepSize) * stepSize;
+	        }
+			value = Math.min(Math.max(value, minimum),maximum);
+	        spinner.setValue(value);
+		}
+	}
+
+    /**
+     * An editor for a <code>JSpinner</code> whose model is a
+     * <code>SpinnerIntModel</code>.  The value of the editor is
+     * displayed with a <code>JFormattedTextField</code> whose format
+     * is defined by a <code>NumberFormatter</code> instance whose
+     * <code>minimum</code> and <code>maximum</code> properties
+     * are mapped to the <code>SpinnerIntModel</code>.
+     */
+    public static class IntegerEditor extends DefaultEditor
+    {
+        // This is here until DecimalFormat gets a constructor that
+        // takes a Locale: 4923525
+        private static String getDefaultPattern(Locale locale) {
+            // Get the pattern for the default locale.
+            ResourceBundle rb = LocaleData.getNumberFormatData(locale);
```

Comment:
@skeptical I think ```LocaleData``` has created enough problems now. I think I need to find a more robust way of doing that than what swing does (the swing libs are probably precompiled so it's not a problem when using them right?). The task to be solved is very simple in fact: the OS' locale is fetched and the number format is derived/extracted from it, so that the spinner can format the numbers correctly (different locales have different number separators and decimal separators). It's simply ridiculous that such a simple task should create so many problems, there must be a better way to come up with the correct number format. I didn't look for it, because I deemed it safest to do it the same way swing does, but I've changed my mind now.